### DISCUSSION TITLE
Create/update workflows using  REST api

### DISCUSSION THREAD

**Discussion by ramprakashn** [codingjoe/joeflow#49] [Q&A]
I have gone through the examples in the test app and understand that the joeflow workflows are template driven. Is it possible to drive the workflows using REST api without using templates.

**Comment by codingjoe** [comment]
Hi @ramprakashn,

Thank you for reaching out. That is an excellent question. So in theory, there is no issue creating a task via DRM. However, I think it's important to understand the difficulties before doing so.

A RESTful interface MUST allow object modification. However, joeflow workflows may intentionally only be modified via tasks. Those two concepts don't work well together, since they are so fundamentally different.

All that being said, the solution should be simple. You can either just start a machine task, from within your DRF view. Or try to inherit from both DRFs `APIView` and joeflow's `TaskViewMixin`. In your view, you will need to create the Workflow instance and call `next_task` before you return the response.

I didn't do it myself yet, because of the issues I described above, but please don't hesitate to share some code, should you get stuck somewhere.

Best,
Joe
